### DISCUSSION TITLE
Is it possible to add launch flags like `--disable-web-security` to browser mode launched with the playwright provider

### DISCUSSION THREAD

**Discussion by spookyuser** [vitest-dev/vitest#4866] [Q&A]
First of all thank you sm for browser mode, it is making so much of my testing easier and it basically feels like something i've imagined forever and now exists!!

just wondering if it's possible to add any playwright browser config to the browser that launches, specifically i'd love to be able to toggle --disable-web-security and other arbitrary tags you can control in the playwright launchargs object?

Thanks!

**Comment by sheremet-va** [comment]
You can configure playwright via [`browser.providerOptions`](https://vitest.dev/config/#browser-provideroptions):

```ts
export default defineConfig({
  test: {
    browser: {
      provider: 'playwright',
      providerOptions: {
        launch: {
          args: ['--disable-web-security'],
        }
      }
    }
  }
})
```

**Reply by spookyuser** [reply]
omg thank you!

**Reply by edwincarlflores** [reply]
Hi @sheremet-va @spookyuser , it seems that `--disable-web-security` flag will only work if passed together with the `--user-data-dir="/path/to/dir'`. However, I am receiving this error which I believe is from playwright:
```
Error: browserType.launch: Pass userDataDir parameter to 'browserType.launchPersistentContext(userDataDir, options)' instead of specifying '--user-data-dir' argument
```

<img width="1551" alt="Screenshot 2024-07-23 at 7 58 37 PM" src="https://github.com/user-attachments/assets/bd85ded2-878d-4f2a-8c4d-cf231e3d0874">

Is there a possible workaround for this? I am trying to access an element inside an iframe and I am having cross-origin issues which is expected. `--disable-web-security` should allow me to access the content document of the iframe.

**Reply by sheremet-va** [reply]
Vitest only uses `launch` command, not `launchPersistentContext`. You need to open a feature request for it to be implemented.

**Comment by cb109** [comment]
For anyone trying to achieve the same thing on the python/pytest side of things, this finally worked for me:

```py
@pytest.fixture(scope="session")
def browser(launch_browser):
    browser = launch_browser(
        args=["--disable-web-security"],
    )
    yield browser
    browser.close()
```
